### PR TITLE
Changed '.sum(&block)' to 'to_a.sum(&block)' in spec files

### PR DESCRIPTION
**Pull request by jhsu802701**
For the files within the spec directory, this pull request addresses the following deprecation message:
```
DEPRECATION WARNING: Calling #sum with a block is deprecated and will be removed in Rails 4.1. If you want to perform sum calculation over the array of elements, use `to_a.sum(&block)`.
```
I know that there are numerous instances of these deprecation messages, but we have to start somewhere.

### CODE REVIEW

**Comment by luisramos0** [comment]
Hi Jason, thanks for the PR but most of this work is already done as part of the rails 4-1 upgrade in https://github.com/openfoodfoundation/openfoodnetwork/pull/5919
We can block this PR until https://github.com/openfoodfoundation/openfoodnetwork/pull/5919 is merged and then we see if there's anything here that is not already in 5919 :+1:

**Comment by RachL** [comment]
@luisramos0 I see #5919 is merge. Is there any action to do here?

**Comment by luisramos0** [comment]
this needs a rebase and a green build before code review.

**Comment by RachL** [comment]
@openfoodfoundation/core-devs anyone available to move this one forward? (rebase or close if we already applied the changes)

File: spec/support/performance_helper.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def multi_benchmark(num_samples, cache_key_patterns: [])
         result.total
       end.drop(1) # Do not return the first sample
 
-      avg = (results.sum / results.count * 1000).round(0)
+      avg = (results.to_a.sum / results.count * 1000).round(0)
```
Comment (andrewpbrett) [inline spec/support/performance_helper.rb:17]: Do we need to add the `to_a` here, if we're not passing a block?

File: spec/lib/open_food_network/group_buy_report_spec.rb [hunk 2]
```diff
@@ -56,8 +56,8 @@ module OpenFoodNetwork
 
       line_items = @orders.map(&:line_items).flatten.select{ |li| li.product.supplier == @supplier1 && li.variant == @variant1 }
 
-      sum_quantities = line_items.map(&:quantity).sum
-      sum_max_quantities = line_items.map { |li| li.max_quantity || 0 }.sum
+      sum_quantities = line_items.map(&:quantity).to_a.sum
+      sum_max_quantities = line_items.map { |li| li.max_quantity || 0 }.to_a.sum
```
Comment (andrewpbrett) [inline spec/lib/open_food_network/group_buy_report_spec.rb:60]: Do we need to add the `to_a` here, if we're not passing a block?

**Review by andrewpbrett — CHANGES_REQUESTED** [review]
Hi @jhsu802701 - I rebased this and it's back to mergeable with master. Are you still seeing the deprecation messages though? On the latest build of master, I don't see these particular ones (there are still many others though). 

Should we close this one and #6168 ?

File: spec/lib/open_food_network/group_buy_report_spec.rb (hunk 2 above)
Comment (sauloperez) [reply spec/lib/open_food_network/group_buy_report_spec.rb:60]: this `to_a` seems pointless to me. `map` returns an array already.

**Comment by RachL** [comment]
@andrewpbrett I think we can close them by now, no?

**Comment by Matt-Yorkley** [comment]
We dealt with this issue in a few other PRs. I just double-checked a few of these specs and they currently are not producing warnings in master :+1:

**Pull request closed without merging**